File: src/main/java/org/example/PartitionerConsumer.java
package org.example;

import org.apache.kafka.clients.consumer.*;
import org.apache.kafka.common.serialization.StringDeserializer;

import java.time.Duration;
import java.util.Collections;
import java.util.Properties;

public class PartitionerConsumer {
    public static void main(String[] args) {
        subscribeOurMessages();
    }

    private static void subscribeOurMessages() {

        // consumer
        Consumer consumer = createConsumer();

        // subscribe
        consumer.subscribe(Collections.singletonList("partitioner-topic-test"));

        // messages records
        ConsumerRecords<String, String> poll = consumer.poll(Duration.ofMillis(10000));
        for (ConsumerRecord<String, String> consumerRecord : poll) {
            System.out.println(consumerRecord.value());
        }
    }

    private static Consumer createConsumer() {
        Properties properties = new Properties();

        properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
        properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
        properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
        properties.put(ConsumerConfig.GROUP_ID_CONFIG, "group-id-conf-1");
        properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");

        return new KafkaConsumer<>(properties);
    }
}

File: python/PartitionerConsumer.py
from confluent_kafka import Consumer


class PartitionerConsumer:

    __TOPIC_TO_SUBSCRIBE = "partitioner-topic-test"

    @staticmethod
    def subscribe_our_messages():
        # consumer
        consumer = PartitionerConsumer.create_consumer()

        # subscribe
        consumer.subscribe([PartitionerConsumer.__TOPIC_TO_SUBSCRIBE])

        # messages records
        while True:
            message = consumer.poll(10.0)

            if message is None:
                break

            print(message.value().decode('utf-8'))

        consumer.close()

    @staticmethod
    def create_consumer() -> Consumer:
        return Consumer(
            {
                'bootstrap.servers': 'localhost:9092',
                'group.id': 'group-id-conf-1-python',
                'auto.offset.reset': 'earliest'
            }
        )


def main():
    PartitionerConsumer.subscribe_our_messages()


if __name__ == '__main__':
    main()
